index.js

```diff
@@ -75,7 +75,9 @@ function fastifyCaching (instance, options, next) {
     }
 
     instance.addHook('onRequest', (req, res, next) => {
-      res.header('Cache-control', value)
+      if (!res.getHeader('Cache-control')) {
```

Comment:
Is it accurate to overwrite the header if it has been set to an empty string?

Reply:
Good question, I don't know enough about the fastify ecosystem and common patterns to give an answer.

I would personally give the control back to users with a dedicated option and a good default value tbh (2nd proposition [in the issue](https://github.com/fastify/fastify-caching/issues/14#issuecomment-1467112296))

Reply:
Let's use `res.hasHeader` instead

Reply:
deal 👍